src/environments.js

```diff
@@ -34,6 +34,12 @@ function parseArray(parser, result) {
                                  parser.lexer, pos);
         }
     }
+    // Remove trailing line break, if there is one
+    var lastRow = body[body.length - 1];
+    if (body.length > 1 && lastRow.length === 1
+        && lastRow[0].value.length === 0) {
+        body.pop();
+    }
```

Comment:
Inside the macro for matrices, TeX leaves a `\crcr` for the last row which acts like a `\cr`, unless it follows a `\cr` in which case it does nothing.  This is why the last `\cr` or `\\` is optional with matrices.  But this also means it trims at most the last row.  See this [reference](https://www.tug.org/utilities/plain/cseq.html#crcr-rp)